GoRogue/MapGeneration/Region.cs

```diff
@@ -1,355 +1,478 @@
-        /// with the imposing region
+        /// Moves the region in the indicated direction.
         /// </summary>
-        /// <param name="imposing">the region to check for common outer points</param>
-        public void RemoveOverlappingPoints(Region imposing)
+        /// <param name="by">The amount (X and Y) to translate this region by.</param>
+        /// <returns>A new, translated region</returns>
+        public virtual Region Translate(Point by)
         {
-            foreach (Point c in imposing.Points)
-            {
-                while (_outerPoints.Contains(c))
-                    _outerPoints.Remove(c);
-                while (_innerPoints.Contains(c))
-                    _innerPoints.Remove(c);
-            }
+            var nw = NorthWestCorner + by;
+            var ne = NorthEastCorner + by;
+            var se = SouthEastCorner + by;
+            var sw = SouthWestCorner + by;
+            return new Region(nw, ne, se, sw);
         }
-
         /// <summary>
         /// Rotates a region around it's center.
         /// </summary>
         /// <param name="degrees">The amount of degrees to rotate this region</param>
-        /// <returns>A region equal to the original region rotated by the given degree</returns>
-        public virtual Region Rotate(double degrees) => Rotate(degrees, Center);
+        /// <returns>A region, equal to the original region rotated by the given degree</returns>
+        public virtual Region Rotate(double degrees)
+            => Rotate(degrees, Center);
 
         /// <summary>
         /// Rotates this region by an arbitrary number of degrees
         /// </summary>
         /// <param name="degrees">The amount of degrees to rotate this region</param>
         /// <param name="origin">The Point around which to rotate</param>
-        /// <returns>This region, rotated.</returns>
-        /// <remarks>
-        /// This is destructive to the region's <see cref="Connections"/>, so you should try to refrain from generating
-        /// those until after you've performed your rotations.
-        /// </remarks>
+        /// <returns>A region, equal to the original region rotated by the given degree</returns>
         public virtual Region Rotate(double degrees, Point origin)
         {
-            degrees = MathHelpers.WrapAround(degrees, 360);
+            degrees %= 360;
 
             //figure out the new corners post-rotation
-            List<Point> corners = new List<Point>();
-            Point southwest = SouthWestCorner.Rotate(degrees, origin);
-            corners.Add(southwest);
-            Point southeast = SouthEastCorner.Rotate(degrees, origin);
-            corners.Add(southeast);
-            Point northwest = NorthWestCorner.Rotate(degrees, origin);
-            corners.Add(northwest);
-            Point northeast = NorthEastCorner.Rotate(degrees, origin);
-            corners.Add(northeast);
+            Point[] corners = new []
```

Comment:
NOTE: I've got some simple optimizations in mind for this but may be best to wait until the future of corners is decided.